core/server/policies.go

```diff
@@ -0,0 +1,248 @@
+		opts = append(opts, client.Limit(m.Pagination.PageSize))
+		opts = append(opts, client.Continue(m.Pagination.PageToken))
+	}
+
+	var continueToken string
+	var listsV2beta2 map[string][]client.ObjectList
+
+	if m.ClusterName == "" {
+		clistV2beta2 := clustersmngr.NewClusteredList(func() client.ObjectList {
+			return &pacv2beta2.PolicyList{}
+		})
+
+		var errsV2beta2 clustersmngr.ClusteredListError
+
+		if err := clustersClient.ClusteredList(ctx, clistV2beta2, false, opts...); err != nil {
+			if !errors.As(err, &errsV2beta2) {
+				return nil, fmt.Errorf("error while listing v2beta2 policies: %w", err)
+			}
+		}
+		for _, e := range errsV2beta2.Errors {
+			if !strings.Contains(e.Err.Error(), "no matches for kind \"Policy\"") {
+				respErrors = append(respErrors, &pb.ListError{ClusterName: e.Cluster, Message: e.Err.Error()})
+			}
+		}
+
+		continueToken = clistV2beta2.GetContinue()
+		listsV2beta2 = clistV2beta2.Lists()
+	} else {
+		listV2beta2 := &pacv2beta2.PolicyList{}
+
+		if err := clustersClient.List(ctx, m.ClusterName, listV2beta2, opts...); err != nil {
+			return nil, fmt.Errorf("error while listing policies for cluster %s: %w", m.ClusterName, err)
+		}
+
+		continueToken = listV2beta2.GetContinue()
+		listsV2beta2 = map[string][]client.ObjectList{m.ClusterName: {listV2beta2}}
+	}
+
+	var policies []*pb.Policy
+	collectedPolicies := map[string]struct{}{}
+	for clusterName, lists := range listsV2beta2 {
+		for _, l := range lists {
+			list, ok := l.(*pacv2beta2.PolicyList)
+			if !ok {
+				continue
+			}
+			for i := range list.Items {
+				policy, err := toPolicyResponse(list.Items[i], clusterName, false)
+				if err != nil {
+					return nil, err
+				}
+
+				policies = append(policies, policy)
+				collectedPolicies[getClusterPolicyKey(clusterName, list.Items[i].GetName())] = struct{}{}
+			}
+		}
+	}
+
+	return &pb.ListPoliciesResponse{
+		Policies:      policies,
+		Total:         int32(len(policies)),
+		NextPageToken: continueToken,
+		Errors:        respErrors,
+	}, nil
+}
+
+func (cs *coreServer) GetPolicy(ctx context.Context, m *pb.GetPolicyRequest) (*pb.GetPolicyResponse, error) {
+	clustersClient, err := cs.clustersManager.GetImpersonatedClientForCluster(ctx, auth.Principal(ctx), m.ClusterName)
+	if err != nil {
+		return nil, fmt.Errorf("error getting impersonating client: %w", err)
+	}
+
+	if m.ClusterName == "" {
+		return nil, errRequiredClusterName
+	}
+	policyCRv2beta2 := pacv2beta2.Policy{}
+
+	if err := clustersClient.Get(ctx, m.ClusterName, types.NamespacedName{Name: m.PolicyName}, &policyCRv2beta2); err != nil {
+		return nil, fmt.Errorf("error while getting policy %s from cluster %s: %w", m.PolicyName, m.ClusterName, err)
+	}
+
+	var policy *pb.Policy
+
+	policy, err = toPolicyResponse(policyCRv2beta2, m.ClusterName, true)
+	if err != nil {
+		return nil, err
+	}
+
+	return &pb.GetPolicyResponse{Policy: policy}, nil
+}
+
+func getClusterPolicyKey(clusterName, policyId string) string {
```

Comment:
this is only used in collectedPolicies and I don't see collectedPolicies being used anywhere, shouldn't they both be removed?